### ISSUE TITLE
Missing TLS HandshakeType support (certificate_request)

### ISSUE THREAD

**Issue by richard-powers** [ziglang/zig#19521]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

I found this issue when trying to connect to a postgres database which requires ssl.
In [std.crypto.tls.Client](https://github.com/ziglang/zig/blob/0.11.x/lib/std/crypto/tls/Client.zig#L480) there is a switch statement over the HandshakeType, but `certificate_request` is unhandled, so I simply receive `error.TlsUnexpectedMessage` instead of being able to connect to my database.

In the [TLS RFC](https://www.rfc-editor.org/rfc/rfc8446#section-4.3.2), `A server which is authenticating with a certificate MAY optionally request a certificate from the client`

So it seems this must be supported in some cases, unless I'm mistaken?

### Expected Behavior

`HandshakeType.certificate_request` to be supported, if it is indeed required for the protocol

**Comment by clickingbuttons** [comment]
Yes, the TLS Client should add an option for a Certificate to respond to these messages. This isn't required for HTTPS, which is what the client is currently designed for.

I think this is blocked on #14171. After it lands it this feature should be added to both the TLS client and the TLS server and tested.

**Comment by rockorager** [comment]
Chiming in here that this is a blocker for me in an IRC client appllication. Several IRC servers fail TLS handshakes because of this.

**Comment by VaskoBozhurski** [comment]
Just wanted to note that some HTTPS endpoints such as the Kubernetes API Server may do a `certificate_request` during the handshake.

**Comment by GrayHatter** [comment]
This prevents me from using srctree as source for `build.zig.zon` dependencies. 

validation tests https://gist.github.com/GrayHatter/12148218f141041a105724010763d308

example of missing support https://srctree.gr.ht/repo/zmenu/ref/de2d81d0a1baea47c710d55a8e50262eec29139f/blob/build.zig.zon#L9
